### PR TITLE
Misc: Cleanup unused code, warnings, constants, initializations, declarations.

### PR DESCRIPTION
**Pull request by lightningterror**
### Description of Changes

Misc: Cleanup unused code, warnings, constants, initializations, declarations.
### Rationale behind Changes

Cleaner code.
### Suggested Testing Steps

CI builds.

### CODE REVIEW

File: pcsx2/SPR.cpp [hunk 1]
```diff
@@ -179,14 +179,14 @@ int  _SPR0chain()
...
-	int cycles = _SPR0chain() * BIAS;
+	const int cycles = _SPR0chain() * BIAS;
 	CPU_INT(DMAC_FROM_SPR, cycles);
 }
 
 void _SPR0interleave()
 {
 	int qwc = spr0ch.qwc;
-	int sqwc = dmacRegs.sqwc.SQWC;
+	const int sqwc = dmacRegs.sqwc.SQWC;
```
Comment (refractionpcsx2) [inline pcsx2/SPR.cpp:189]: Can you not rearrange these to not be staggered? just looks ugly :P maybe put it 3rd?

File: pcsx2/SPR.cpp [hunk 2]
```diff
@@ -414,7 +414,7 @@ __fi void SPR1chain()
 void _SPR1interleave()
 {
 	int qwc = spr1ch.qwc;
-	int sqwc = dmacRegs.sqwc.SQWC;
+	const int sqwc = dmacRegs.sqwc.SQWC;
```
Comment (refractionpcsx2) [inline pcsx2/SPR.cpp:417]: as above

File: pcsx2/Gif.cpp [hunk 3]
```diff
@@ -553,27 +553,24 @@ static __fi bool mfifoGIFrbTransfer()
...
 
 	gifch.madr = dmacRegs.rbor.ADDR + (gifch.madr & dmacRegs.rbsr.RMSK);
 	gifch.tadr = dmacRegs.rbor.ADDR + (gifch.tadr & dmacRegs.rbsr.RMSK);
 
 	if (needWrap && transferred == MFIFOUntilEnd)
 	{
 		// Need to do second transfer to wrap around
-		u32 transferred2;
-		uint secondTransQWC = qwc - MFIFOUntilEnd;
+		const uint secondTransQWC = qwc - MFIFOUntilEnd;
```
Comment (refractionpcsx2) [inline pcsx2/Gif.cpp:567]: could be moved after the src check, just for scope/pointless initialisation sake.

File: pcsx2-qt/Settings/GameCheatSettingsWidget.cpp [hunk 4]
```diff
@@ -167,7 +167,6 @@ void GameCheatSettingsWidget::reloadList()
 		const bool enabled =
 			(std::find(m_enabled_patches.begin(), m_enabled_patches.end(), pi.name) != m_enabled_patches.end());
 
-		const std::string_view name_part = pi.GetNamePart();
```
Comment (refractionpcsx2) [inline pcsx2-qt/Settings/GameCheatSettingsWidget.cpp:170]: Check with sten, not sure if that was accidental or not.

Comment (lightningterror) [reply pcsx2-qt/Settings/GameCheatSettingsWidget.cpp:170]: @stenzek you have been summoned.

Comment (stenzek) [reply pcsx2-qt/Settings/GameCheatSettingsWidget.cpp:170]: It's fine

**Pull request merged**